### PR TITLE
PIPELINE: Fix config bug

### PR DESCRIPTION
**Pull request by nielsboecker**
Boolean and numerical values were parsed as strings, d'oh

### CODE REVIEW

**Comment by nbckr** [comment]
Note: There are some other places in the code where I read the numeric variables and cast them, which is now redundant. I will fix this in the othher branch I'm working on atm. I excluded it here as I made some other changes in these files and wanted to keep this PR here small.

File: HoloPipelines/jobs/jobs_state.py [hunk 1]
```diff
@@ -34,32 +41,33 @@
 )
 
 
-def update_job_state(job_id: str, new_state: str, logger=logging):
+def update_job_state(job_id: str, new_state: str, logger=logging, new: bool = False):
     """
-    Updates the global dictionary that keeps track of all jobs. Note that new_state
+    Updates the state for aa job by updating the job.state file. Note that new_state
```
Comment (boonwj) [inline HoloPipelines/jobs/jobs_state.py:46]: typo "a"

File: HoloPipelines/jobs/jobs_io.py [hunk 2]
```diff
@@ -10,55 +10,106 @@
...
+def get_all_job_subdirectory_names() -> List[str]:
+    return [dir.name for dir in os.scandir(jobs_root) if dir.is_dir()]
+
+
+def read_state_file_for_job(jod_id: str) -> Tuple[str, float]:
+    """
+    Reads state file and returns state and time in seconds since last modification.
+    """
+    state_file_path = get_state_file_path_for_job(jod_id)
+    with open(state_file_path, "r") as state_file:
```
Comment (boonwj) [inline HoloPipelines/jobs/jobs_io.py:96]: There needs to be error handling here, as querying a non-existent job id will throw an exception

File: HoloPipelines/jobs/jobs_io.py [hunk 3]
```diff
@@ -80,16 +131,28 @@ def get_logger_for_job(job_id: str):
...
+    """
+    Returns the path to a log file. Looks for current jobs first. If nothing found,
+    looks for finished jobs. If still not found, returns None.
+    """
+    path = get_directory_path_for_job(job_id)
+    if os.path.isdir(path):
+        return f"{path}/job.log"
+    elif os.path.isdir(path.replace(jobs_root, finished_jobs_root)):
+        return f"{path.replace(jobs_root, finished_jobs_root)}/job.log"
+
```
Comment (boonwj) [inline HoloPipelines/jobs/jobs_io.py:144]: An explicit `return None` will be idiomatic given that the function returns different values and None is one of them

File: HoloPipelines/jobs/jobs_io.py [hunk 4]
```diff
@@ -10,55 +10,106 @@
...
+    return os.path.exists(get_state_file_path_for_job(job_id))
+
+
+def write_state_file_for_job(jod_id: str, state: str) -> None:
+    state_file_path = get_state_file_path_for_job(jod_id)
+    with open(state_file_path, "w") as state_file:
+        state_file.write(state)
 
 
 def get_logger_for_job(job_id: str):
```
Comment (boonwj) [inline HoloPipelines/jobs/jobs_io.py:115]: Optional: type hints for return value?

File: HoloPipelines/jobs/jobs_state.py [hunk 5]
```diff
@@ -5,17 +5,24 @@
...
+    get_all_job_subdirectory_names,
+    read_state_file_for_job,
+    remove_temporary_data_for_job,
+    remove_log_file_for_job,
+    move_job_to_finished_jobs_directory,
+    write_state_file_for_job,
+    state_file_for_job_exists,
+)
 
 JobState = Enum(
```
Comment (boonwj) [inline HoloPipelines/jobs/jobs_state.py:27]: `JobState` does not feel like a proper python variable naming convention for global variables. The naming convention should be similar to python functions.

https://www.python.org/dev/peps/pep-0008/#global-variable-names

**Review by boonwj — APPROVED** [review]
No reason to reject the config bug fixes as this is part of the major refactoring changes. 

In addition, this PR also introduces changes to the way job states are tracked and managed.

It's best to merge it with the major-refactor branch and be evaluated as a whole.

File: HoloPipelines/jobs/jobs_io.py (hunk 2 above)
Comment (nielsboecker) [reply HoloPipelines/jobs/jobs_io.py:96]: Agree, did it in the other branch

File: HoloPipelines/jobs/jobs_state.py (hunk 5 above)
Comment (nielsboecker) [reply HoloPipelines/jobs/jobs_state.py:27]: Hm but it is an Enum right? See for instance how they use it in the docs: https://docs.python.org/3/library/enum.html#functional-api

**Pull request merged**